### PR TITLE
store: introduce cold storage

### PR DESCRIPTION
**Pull request by mina86**
Add a `cold_store` Cargo feature which enables the option to configure
the node with cold storage.  At the moment, all this does is open the
cold database and doesn’t enable any other features.  The idea is that
this can now allow experimenting with code that needs access to the
cold storage.

### CODE REVIEW

**Comment by akhi3030** [comment]
@posvyatokum: I'm happy to approve once your reviews are in and you approve.

File: core/store/src/opener.rs [hunk 1]
```diff
@@ -25,6 +23,21 @@ pub enum StoreOpenerError {
     #[error("Database already exists")]
     DbAlreadyExists,
 
+    /// Hot database exists but cold doesn’t or the other way around.
+    #[error("Hot and cold databases must either both exist or not")]
+    HotColdExistenceMismatch,
+
+    /// Hot database exists but cold doesn’t or the other way around.
```
Comment (posvyatokum) [inline core/store/src/opener.rs:30]: Wrong comment (copy from previous error)

File: core/store/src/opener.rs [hunk 2]
```diff
@@ -25,6 +23,21 @@ pub enum StoreOpenerError {
...
+    #[error(
+        "Hot database version ({hot_version}) doesn’t match \
+         cold databases version ({cold_version})"
+    )]
+    HotColdVersionMismatch { hot_version: DbVersion, cold_version: DbVersion },
+
+    /// Hot database exists but cold doesn’t or the other way around.
```
Comment (posvyatokum) [inline core/store/src/opener.rs:37]: Same here, wrong comment (copy from HotColdExistenceMismatch)

File: nearcore/src/lib.rs [hunk 3]
```diff
@@ -66,6 +73,23 @@ fn open_storage(home_dir: &Path, near_config: &mut NearConfig) -> anyhow::Result
         Err(StoreOpenerError::DbDoesNotExist) => unreachable!(),
         // Cannot happen with Mode::ReadWrite
         Err(StoreOpenerError::DbAlreadyExists) => unreachable!(),
+        Err(StoreOpenerError::HotColdExistenceMismatch) => {
+            Err(anyhow::anyhow!(
+                "Hot and cold databases must but exist or both not exist.\n\
```
Comment (posvyatokum) [inline nearcore/src/lib.rs:78]: but -> both?

**Review by posvyatokum — APPROVED** [review]
Code looks good, all my comments are about comments.

**Review by akhi3030 — APPROVED** [review]

**Pull request merged**